app/questionnaire/questionnaire_schema.py

```diff
@@ -94,11 +95,39 @@ def __init__(
         self._populate_answer_dependencies()
         self._populate_when_rules_section_dependencies()
         self._populate_calculated_summary_section_dependencies()
+        self._populate_min_max_map()
 
     @cached_property
     def answer_dependencies(self) -> ImmutableDict[str, set[AnswerDependent]]:
         return ImmutableDict(self._answer_dependencies_map)
 
+    def _populate_min_max_map(self) -> None:
+        for answer_id, answer in self._answers_by_id.items():
+            if (answer_type := answer[0].get("type")) and answer_type not in [
```

Comment:
this works for answers that are 1 reference deep, to include others, maybe you could have a populate minmax for answer function similar to the filter one and do something like this
```
if value := answer.get(min_max, {}).get("value"):
    if isinstance(value, int):
        self.min_and_max_map[answer["id"]] = value
    elif isinstance(value, dict):
        if value.get("source") == "answers":
            self.min_and_max_map[answer["id"]] = self.min_and_max_map[value["identifier"]]
```
which might make other part simlar, or if having every answer is overkill, skip the int case